Generators/src/FlowMapper.cxx

```diff
@@ -0,0 +1,155 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Generators/FlowMapper.h"
+#include "TH1D.h"
+#include "TH3D.h"
+#include "TF1.h"
+#include <fairlogger/Logger.h>
+
+namespace o2
+{
+namespace eventgen
+{
+
+/*****************************************************************/
+/*****************************************************************/
+
+FlowMapper::FlowMapper()
+{
+  // base constructor. Creates cumulative function only so that's already in place but not much else.
+  fCumulative = new TF1("fCumulative", "x+[0]*TMath::Sin(2*x)", 0, 2 * TMath::Pi());
+  hv2vsPt = 0x0;
+  hEccVsB = 0x0;
+  hLUT = 0x0;
+
+  binsPhi = 4000; // a first guess
+}
+
+void FlowMapper::Setv2VsPt(TH1D* hv2VsPtProvided)
+{
+  // Sets the v2 vs pT to be used.
+  hv2vsPt = hv2VsPtProvided;
+}
+void FlowMapper::SetEccVsB(TH1D* hEccVsBProvided)
+{
+  // Sets the v2 vs pT to be used.
+  hEccVsB = hEccVsBProvided;
+}
+void FlowMapper::CreateLUT()
+{
+  if (!hv2vsPt) {
+    LOG(fatal) << "You did not specify a v2 vs pT histogram!";
+    return;
+  }
+  if (!hEccVsB) {
+    LOG(fatal) << "You did not specify an ecc vs B histogram!";
+    return;
+  }
+  LOG(info) << "Proceeding to creating a look-up table...";
+  const Long_t nbinsB = hEccVsB->GetNbinsX();
+  const Long_t nbinsPt = hv2vsPt->GetNbinsX();
+  const Long_t nbinsPhi = binsPhi; // constant in this context necessary
+  Double_t binsB[nbinsB + 1], binsPt[nbinsPt + 1], binsPhi[nbinsPhi + 1];
+
+  for (int ii = 0; ii < nbinsB + 1; ii++)
+    binsB[ii] = hEccVsB->GetBinLowEdge(ii + 1);
+  for (int ii = 0; ii < nbinsPt + 1; ii++)
+    binsPt[ii] = hv2vsPt->GetBinLowEdge(ii + 1);
+  for (int ii = 0; ii < nbinsPhi + 1; ii++)
```

Comment:
In O2, The CI will check for enclosing braces `{}` even in one-line if statements or loops.

Reply:
Ah thanks, ok, changed !